### ISSUE TITLE
Inquiry regarding GroundingDINO pre-trained weights and performance reproduction gap

### ISSUE THREAD

**Issue by SuperLuJC** [Ideal-ljl/AerialVG#12]
Hi authors,

Thank you for your great work on AerialVG.

I have a question regarding the training strategy mentioned in the paper. specifically, did you use the pre-trained weights from GroundingDINO-Large or GroundingDINO-Tiny?

Case 1: If you used GroundingDINO-Large: Since the official GroundingDINO repository has not released the corresponding checkpoints for the Large version publicly, could you kindly provide the weight file you used?

Case 2: If you used GroundingDINO-Tiny (Swin-T): I attempted to reproduce the results on the AerialVG dataset using the official Swin-T weights. However, there is a significant performance gap:

My reproduction: Top-1 is around 30%, and Top-5 reaches at most 68%.

Paper results: Top-5 is reported as 78%.

Could you please provide some guidance on this discrepancy?

Looking forward to your reply. Thanks!

**Comment by Ideal-ljl** [comment]
I used GroundingDINO-Tiny , and the ckpt is provide bt opengroundingdino, this is the link:[https://github.com/longzw1997/Open-GroundingDino/releases/download/v0.1.0/gdinot-1.8m-odvg.pth](url), Did you use the same ckpt?

**Comment by SuperLuJC** [comment]
Hi author,

Thanks for the quick clarification and the link!

Regarding the checkpoint:

I was previously using the official groundingdino_swint_ogc.pth from the IDEA-Research repository, which likely caused the performance gap. I will download the gdinot-1.8m-odvg.pth you provided and retrain the model immediately.

However, I have a follow-up question regarding the paper content:

I noticed that in the comparison table (as shown in the attached screenshot), the backbone for "AerialVG (Ours)" is listed as Swin-L.

Could you please clarify if Swin-L was involved in the specific training pipeline of AerialVG?

If yes, in which step was it used? 

Regarding the main result on the AerialVG dataset (Top-5 ~78%): Was this achieved purely with the Swin-Tiny checkpoint you just shared, or does it rely on a Swin-Large backbone?

I want to ensure I fully understand the model architecture to correctly reproduce the reported performance.

Thanks again for your help!

**Comment by Ideal-ljl** [comment]
You can get the result through the ckpt by the links with Swin -T. The results of GroundingDINO with Swin-L in the Table 2 is  provided by the  GroundingDINO. And the results of AerialVG in the Table2 is the AerialVG model initialization with Swin transformers Large with is provide by official Swin transformers.

**Comment by SuperLuJC** [comment]
Got it! That makes everything clear.

So, the Table 2 (RefCOCO) results used Swin-L for comparison, but for the AerialVG dataset performance I am reproducing, the Swin-Tiny checkpoint you provided (gdinot-1.8m-odvg.pth) is sufficient to achieve the reported results.

I will proceed with the training using the provided Swin-Tiny weights.

Thank you so much for your patience and detailed explanation!

**Comment by SuperLuJC** [comment]
Hi author,
I have evaluated this checkpoint to verify the performance, and I have some questions regarding its usage and the data reported in the paper's comparison table.
Clarification on the usage of gdinot-1.8m-odvg.pth I ran a direct evaluation (eval) on this checkpoint (without training) and observed the following results:
Top-1: ~18.1%
Top-5: ~43.6%
discrepancy between my evaluation and the "Grounding DINO" baseline reported in the paper (as shown in the attached screenshot):
Paper's Zero-Shot (Grounding DINO): Top-5 is 13.79%.
My Zero-Shot (using gdinot-1.8m-odvg.pth): Top-5 is ~43.6%.
Could you please clarify the source of the four data points in the "Grounding DINO" row?

<img width="887" height="147" alt="Image" src="https://github.com/user-attachments/assets/b9615dc2-fbcd-4e2b-997a-8a32cbb98c75" />

<img width="875" height="110" alt="Image" src="https://github.com/user-attachments/assets/5336ed23-80d7-436c-a2a8-8f1d4d61d1e8" />

**Comment by Ideal-ljl** [comment]
Thank you for your question. If I remember correctly, it should be this ckpt. I will check this issue this week.

**Comment by SuperLuJC** [comment]
Thanks for your reply and for looking into this!

While you are checking the checkpoint details, could you also kindly share the specific training configuration file used for the AerialVG experiments?

Having the exact hyperparameters (such as learning rate, batch size, scheduler, etc.) would be extremely helpful. This will allow me to ensure my reproduction setup is perfectly aligned with yours and rule out any hyperparameter discrepancies while verifying the checkpoint performance.

Looking forward to your update. Thanks!

**Comment by Ideal-ljl** [comment]
Since I haven't organized these details yet, I can give you a reference first. If I remember correctly, it should be this; you can try it.

**Comment by Ideal-ljl** [comment]
data_aug_scales = [480, 512, 544, 576, 608, 640, 672, 704, 736, 768, 800]
data_aug_max_size = 1333
data_aug_scales2_resize = [400, 500, 600]
data_aug_scales2_crop = [384, 600]
data_aug_scale_overlap = None
batch_size = 8
modelname = 'groundingdino'
backbone = 'swin_T_224_1k'
position_embedding = 'sine'
pe_temperatureH = 20
pe_temperatureW = 20
return_interm_indices = [1, 2, 3]
enc_layers = 6
dec_layers = 6
pre_norm = False
dim_feedforward = 2048
hidden_dim = 256
dropout = 0.0
nheads = 8
num_queries = 900
query_dim = 4
num_patterns = 0
num_feature_levels = 4
enc_n_points = 4
dec_n_points = 4
two_stage_type = 'standard'
two_stage_bbox_embed_share = False
two_stage_class_embed_share = False
transformer_activation = 'relu'
dec_pred_bbox_embed_share = True
dn_box_noise_scale = 1.0
dn_label_noise_ratio = 0.5
dn_label_coef = 1.0
dn_bbox_coef = 1.0
embed_init_tgt = True
dn_labelbook_size = 91
max_text_len = 256
text_encoder_type = "bert-base-uncased"
use_text_enhancer = True
use_fusion_layer = True
use_checkpoint = True
use_transformer_ckpt = True
use_text_cross_attention = True
text_dropout = 0.0
fusion_dropout = 0.0
fusion_droppath = 0.1
sub_sentence_present = True
max_labels = 50                               # pos + neg
lr = 0.001                                   # base learning rate
backbone_freeze_keywords = None               # only for gdino backbone
freeze_keywords = ['bert']                    # for whole model, e.g. ['backbone.0', 'bert'] for freeze visual encoder and text encoder
lr_backbone = 1e-05                           # specific learning rate
lr_backbone_names = ['backbone.0', 'bert']
lr_linear_proj_mult = 1e-05
lr_linear_proj_names = ['ref_point_head', 'sampling_offsets']
weight_decay = 0.00001
param_dict_type = 'ddetr_in_mmdet'
ddetr_lr_param = False
epochs = 15
lr_drop = 4
save_checkpoint_interval = 1
clip_max_norm = 0.1
onecyclelr = False
multi_step_lr = False
lr_drop_list = [4, 8]
frozen_weights = None
dilation = False
pdetr3_bbox_embed_diff_each_layer = False
pdetr3_refHW = -1
random_refpoints_xy = False
fix_refpoints_hw = -1
dabdetr_yolo_like_anchor_update = False
dabdetr_deformable_encoder = False
dabdetr_deformable_decoder = False
use_deformable_box_attn = False
box_attn_type = 'roi_align'
dec_layer_number = None
decoder_layer_noise = False
dln_xy_noise = 0.2
dln_hw_noise = 0.2
add_channel_attention = False
add_pos_value = False
two_stage_pat_embed = 0
two_stage_add_query_num = 0
two_stage_learn_wh = False
two_stage_default_hw = 0.05
two_stage_keep_all_tokens = False
num_select = 300
batch_norm_type = 'FrozenBatchNorm2d'
masks = False
aux_loss = True
set_cost_class = 1.0
set_cost_bbox = 5.0
set_cost_giou = 2.0
cls_loss_coef = 2.0
bbox_loss_coef = 5.0
giou_loss_coef = 2.0
enc_loss_coef = 1.0
interm_loss_coef = 1.0
no_interm_box_loss = False
mask_loss_coef = 1.0
dice_loss_coef = 1.0
focal_alpha = 0.25
focal_gamma = 2.0
decoder_sa_type = 'sa'
matcher_type = 'HungarianMatcher'
decoder_module_seq = ['sa', 'ca', 'ffn']
nms_iou_threshold = -1
dec_pred_class_embed_share = True
match_unstable_error = True
use_ema = False
ema_decay = 0.9997
ema_epoch = 0
use_detached_boxes_dec_out = False
use_coco_eval = True
dn_scalar = 100
relation_num_layers = 3
topk = 15
relation_weight = 0.4

**Comment by SuperLuJC** [comment]
Hi author,

Thank you so much for sharing the detailed configuration parameters! This is extremely helpful for my reproduction.

I noticed the config includes specific settings for the relation module, such as relation_num_layers = 3 and relation_weight = 0.4.

I have one final question regarding the training pipeline/strategy:

When using the provided pre-trained weights (gdinot-1.8m-odvg.pth) with this configuration, should I perform the training in a single stage or multiple stages?

Option A (Direct Fine-tuning): Do I simply load the gdinot-1.8m-odvg.pth weights and start training the entire model (including the relation module) directly for 15 epochs?

Option B (Two-Stage Strategy): Or is there a specific curriculum? For example, do I need to freeze the relation layers first to fine-tune the grounding part, and then unfreeze them for a second stage of training?

I want to make sure I don't miss any specific "freezing/unfreezing" steps that might be required for convergence.

Thanks again for your patience and help!

**Comment by Ideal-ljl** [comment]
I don't remember the specific details very clearly, but my intuition tells me that Option A should work.

**Comment by SuperLuJC** [comment]
Thank you! I appreciate you sharing your intuition and the configuration details.
I will proceed with Option A (Direct Fine-tuning) using the gdinot-1.8m-odvg.pth checkpoint and the parameters you provided.
I’ll give it a try and will reach out again only if I encounter specific issues during the reproduction process.、
Thanks again for all your help!

**Comment by Ideal-ljl** [comment]
Did you install both GroundingDINO and AerialVG in the same virtual environment? This could affect the results.

**Comment by SuperLuJC** [comment]
Yes, I installed both of them in the same virtual environment.
Both the GroundingDINO and AerialVG codebases are running in the exact same environment.

**Comment by Ideal-ljl** [comment]
The reason I'm pointing out this issue is that I did the same thing when I was developing AerialVG, but at that time I found that some settings in AerialVG would override the settings in GroundingDINO (this might be a problem with some of my function naming). I'm not sure if this problem still exists now.

**Comment by SuperLuJC** [comment]
Thanks for the heads-up!

I was indeed running both in the same environment, so it is very likely that the settings conflict you mentioned is causing the issue.

I will follow your advice to create a fresh, isolated virtual environment exclusively for AerialVG and re-run the evaluation.

I'll let you know if this resolves the performance discrepancy. Thanks again for your help!

**Comment by SuperLuJC** [comment]
Hi author,
Following your advice, I created a fresh, isolated environment for AerialVG.
I successfully reproduced the paper's final results using aerialvg.pth: Top-1 50.46% / Top-5 86.53% (matching the paper's 50.01% / 87.00%).
My main question is regarding the significance of the gdinot-1.8m-odvg.pth checkpoint.
I initially thought this checkpoint corresponded to the "Grounding DINO" Zero-Shot row in Table 2 (which reports Top-1 11.10 / Top-5 13.79).
However, when I evaluated gdinot-1.8m-odvg.pth in my clean environment, I got significantly higher results:
My Evaluation (gdinot-1.8m-odvg.pth): Top-1: 17.69%, Top-5: 41.65%
Table 2 "Grounding DINO" (Zero-Shot): Top-1: 11.10%, Top-5: 13.79%
Could you please clarify the meaning of this data?

<img width="1088" height="75" alt="Image" src="https://github.com/user-attachments/assets/11578715-c6a7-4d92-9edc-04f15854cf70" />

<img width="1143" height="75" alt="Image" src="https://github.com/user-attachments/assets/0588b223-e438-4e6f-9884-50188a5081cc" />

<img width="565" height="288" alt="Image" src="https://github.com/user-attachments/assets/28c00fba-ea76-4e86-9853-4258e3371b3c" />

**Comment by Ideal-ljl** [comment]
Thank you for your attention. Logically, the result for "Grounding DINO" in Table 2 should be the result I obtained using gdinot-1.8m-odvg.pth. I'm also a bit confused by this result, but I've been quite busy lately. I'll retest it in a while to find the problem.

**Comment by SuperLuJC** [comment]
Thank you for the update.
I completely understand that you are busy. Please take your time to check this issue when you are available.
Looking forward to hearing from you later!

**Comment by Yikkkkk** [comment]
> Thank you for the update. I completely understand that you are busy. Please take your time to check this issue when you are available. Looking forward to hearing from you later!

Hello, may I ask if you have fully replicated the results presented in the paper? Is it sufficient to just add more training parameters?

**Comment by SuperLuJC** [comment]
First, sincere apologies for the delayed response—I completely missed the email notification for this thread!

To answer your question: unfortunately, no, I haven't been able to fully replicate the results from the paper. I tried various methods and tweaked multiple parameters, but there remained a significant gap in the top metrics, and I couldn't pinpoint the exact cause of the issue.

Because of this, I've actually stopped troubleshooting it for now and have switched to experimenting with a different model instead.

Let me know if you end up making any progress on your end! @Yikkkkk 

> > Thank you for the update. I completely understand that you are busy. Please take your time to check this issue when you are available. Looking forward to hearing from you later!
> 
> Hello, may I ask if you have fully replicated the results presented in the paper? Is it sufficient to just add more training parameters?
